astropy/modeling/parameters.py

```diff
@@ -55,6 +56,52 @@ def _tofloat(value):
     return value
 
 
+# Helpers for implementing operator overloading on Parameter
+
+def _binary_arithmetic_operation(op, right=False):
+    @functools.wraps(op)
+    def wrapper(self, val):
+
+        if self._model is None:
+            return NotImplemented
+
+        if right:
+            return op(val, self.value)
+        else:
+            return op(self.value, val)
+
+    return wrapper
+
+
+def _binary_comparison_operation(op):
+    @functools.wraps(op)
+    def wrapper(self, val):
+
+        if self._model is None:
+            if op is operator.lt:
+                # Because OrderedDescriptor uses __lt__ to work, we need to
+                # call the super method, but only when not bound to an instance
+                # anyways
+                return super(self.__class__, self).__lt__(val)
+            else:
+                return NotImplemented
+
+        return np.all(op(self.value, val))
```

Comment:
Why the `np.all`? This is not in the code this replaces, and also different from the convention for arrays. I think this should just be the same as for other binary operators.